client/state/data-layer/wpcom/concierge/schedules/appointments/book/index.js

```diff
@@ -0,0 +1,104 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { translate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { updateConciergeBookingStatus } from 'state/concierge/actions';
+import { errorNotice } from 'state/notices/actions';
+import { CONCIERGE_APPOINTMENT_CREATE, CONCIERGE_APPOINTMENT_RESCHEDULE } from 'state/action-types';
+import {
+	CONCIERGE_STATUS_BOOKED,
+	CONCIERGE_STATUS_BOOKING,
+	CONCIERGE_STATUS_BOOKING_ERROR,
+	CONCIERGE_ERROR_NO_AVAILABLE_STAFF,
+} from 'me/concierge/constants';
+import fromApi from './from-api';
+import toApi from './to-api';
+import { recordTracksEvent, withAnalytics } from 'state/analytics/actions';
+
+export const bookConciergeAppointment = ( { dispatch }, action ) => {
+	dispatch( updateConciergeBookingStatus( CONCIERGE_STATUS_BOOKING ) );
```

Comment:
It appears like this isn't really part of this refactor so I don't think it's a big point, but I'm guessing that `CONCIERGE_STATUS_BOOKING` doesn't really do anything other than set a boolean flag. If that's the case it's probably not needed. If you are curious and want to circle around after this PR I'd be happy to look into with you.

Reply:
It might be the case that `CONCIERGE_STATUS_BOOKING` is not the best name for this constant but it actually sets the `redux.concierge.status: 'booking';` where `status` is an string enum.